common/persistence/dataInterfaces.go

```diff
@@ -407,8 +407,8 @@ type (
 	// RegisterHistoryTaskReaderRequest is a hint for underlying persistence implementation
 	// that a new queue reader is created by queue processing logic
 	RegisterHistoryTaskReaderRequest struct {
-		// TODO: use shard identity = shardID + owner, instead of just shardID here.
 		ShardID      int32
+		ShardOwner   string
```

Comment:
I can introduce a ShardIdentity struct to persistence interface/package, just seems a bit weird to me as this identity is an application layer concept. Ideally the dependency should be reversed between persistence & business logic.